### DISCUSSION TITLE
ORM proxy classes set to false, but now and then "Unable to create a proxy for a final class" appears

### DISCUSSION THREAD

**Discussion by gchamorro** [symfony/symfony#42861] [Q&A]
I'm fairly new to Symfony.

I'm creating a web application using the Domain Driven Design architecture. The domain models in this architecture are declared as final as a good practice.

Although I turn `auto_generate_proxy_classes` to `false` in my doctrine config file, now and then I get an `"Unable to create a proxy for a final class <My final model>..."`.

I say now and then because it's not a persistent error, so it's quite difficult to me to debug it or find solutions. The stack trace isn't helpful, since it doesn't point to my code but to the index page:

```
  <?php
  use App\Kernel;
  require_once dirname(__DIR__).'/vendor/autoload_runtime.php';
  return function (array $context) {
      return new Kernel($context['APP_ENV'], (bool) $context['APP_DEBUG']);
  };
```

As soon as I refresh the page, the error disappears.

How can I disable the creation of proxy classes for good? Or am I failing to see some misconfiguration?.

I don't want to change my domain to accomodate to the framework (not declaring the models as final), since that's the point of applying the DDD architecture.

**Comment by ahundiak** [comment]
Doctrine uses proxy classes to implement functionality such as lazy loading.  Disabling automatic generation of proxy classes just means it is up to you to manually generate them.  Something which is done with cache:clear.  Doctrine simply won't work without them.

The thing is, under DDD your domain entities are supposed to be strictly business oriented and not have to worry about persistence.  Trying to use Doctrine entities in the domain is basically an error.  There should be another layer that does nothing but persist and hydrate your domain entities.

From a practical point of view,  quite a few DDD apps ignore this sort of separation and just us an ORM directly.  Which means no final classes.  It's actually a bit interesting how many DDD entities end up being anemic.  The choice is yours.
